### ISSUE TITLE
RedundantThis rule proposition

### ISSUE THREAD

**Issue by zielint0** [checkstyle/checkstyle#7895]
Is it possible to detect a redundant "this" modifier?

I would like to have an opposite rule to the "RequireThis" rule.

**Comment by romani** [comment]
I think it is possible, you can try to implement it, testing will show how reliable this Check can be.
If there be to much of false positives, we can always move it where our controversial/experimental/prone-to-false-positve Checks are hosted - https://github.com/sevntu-checkstyle/sevntu.checkstyle

**Comment by zielint0** [comment]
Any good soul to implement it? :-)

**Comment by wilcoln** [comment]
I'm on it.

**Comment by zielint0** [comment]
> I'm on it.

Merci Wilfried! :-)

**Comment by wilcoln** [comment]
@romani, please can you approve issue?

**Comment by wilcoln** [comment]
@romani please can you approve this issue? Issue being not approved is the one reason for CI failure. If issue should not be approved please let me know. Thanks.

**Comment by romani** [comment]
there is almost completed (90%) https://github.com/checkstyle/checkstyle/pull/8013 but abandoned on complete .
code from this PR can be reused by anybody who want to fix this issue.

**Comment by Rahulkhinchi03** [comment]
@romani May I work on this PR?

**Comment by Rahulkhinchi03** [comment]
@romani sir,
May I work on it?

Things I want to clarify before I start?
I have to make 2 PR or 2 commits for?
```
RedundantThis
```
```
RequireThis
```

**Comment by Rahulkhinchi03** [comment]
@romani sir, Can I work on it?

**Comment by rnveach** [comment]
@romani Is there anything stopping this issue from being approved since it is actively being worked on?

**Comment by romani** [comment]
approved

**Comment by Atharv3221** [comment]
I am working on this

**Comment by Atharv3221** [comment]
Anyone working on this can re-use code from https://github.com/checkstyle/checkstyle/pull/19550

**Comment by romani** [comment]
@Carbon14-48 , can you help us here ?

**Comment by M-SaaD-H** [comment]
I'd like to work on this. I'll continue from the implementation in #19550

**Comment by Carbon14-48** [comment]
> [@Carbon14-48](https://github.com/Carbon14-48) , can you help us here ?

yessss i'd love to

**Comment by M-SaaD-H** [comment]
Hi @Carbon14-48,
I'm already working on this and have raised a PR for it.

**Comment by Carbon14-48** [comment]
> Hi [@Carbon14-48](https://github.com/Carbon14-48), I'm already working on this and have raised a PR for it.

hmmmmm i started working it too but it's okay since you raised PR, you finish it off gooood luckk !!!
